openmc/stats/multivariate.py

```diff
@@ -0,0 +1,543 @@
+        Distribution of the azimuthal angle in radians
+
+    """
+
+    def __init__(self, mu=None, phi=None, reference_uvw=[0., 0., 1.]):
+        super().__init__(reference_uvw)
+        if mu is not None:
+            self.mu = mu
+        else:
+            self.mu = Uniform(-1., 1.)
+
+        if phi is not None:
+            self.phi = phi
+        else:
+            self.phi = Uniform(0., 2*pi)
+
+    @property
+    def mu(self):
+        return self._mu
+
+    @property
+    def phi(self):
+        return self._phi
+
+    @mu.setter
+    def mu(self, mu):
+        cv.check_type('cosine of polar angle', mu, Univariate)
+        self._mu = mu
+
+    @phi.setter
+    def phi(self, phi):
+        cv.check_type('azimuthal angle', phi, Univariate)
+        self._phi = phi
+
+    def to_xml_element(self):
+        """Return XML representation of the angular distribution
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing angular distribution data
+
+        """
+        element = ET.Element('angle')
+        element.set("type", "mu-phi")
+        if self.reference_uvw is not None:
+            element.set("reference_uvw", ' '.join(map(str, self.reference_uvw)))
+        element.append(self.mu.to_xml_element('mu'))
+        element.append(self.phi.to_xml_element('phi'))
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate angular distribution from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.stats.PolarAzimuthal
+            Angular distribution generated from XML element
+
+        """
+        mu_phi = cls()
+        params = get_text(elem, 'parameters')
+        if params is not None:
+            mu_phi.reference_uvw = [float(x) for x in params.split()]
+        mu_phi.mu = Univariate.from_xml_element(elem.find('mu'))
+        mu_phi.phi = Univariate.from_xml_element(elem.find('phi'))
+        return mu_phi
+
+
+class Isotropic(UnitSphere):
+    """Isotropic angular distribution.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def to_xml_element(self):
+        """Return XML representation of the isotropic distribution
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing isotropic distribution data
+
+        """
+        element = ET.Element('angle')
+        element.set("type", "isotropic")
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate isotropic distribution from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.stats.Isotropic
+            Isotropic distribution generated from XML element
+
+        """
```

Comment:
As it stands, a user can do a nonsensical thing like `iso = Isotropic(elem = "potato")`, and get a valid Isotropic object (since the `elem` isn't used in the method). They could also pass in a PolarAzimuthal "mu-phi" type elem and still get an isotropic. Though you may anticipate the superclass' version of this method will be used most, this method should still check its inputs. Here's a a couple lines to fix that.

```
distribution = get_text(elem, 'type')
if distribution != "isotropic":
    raise Exception("Expected an XML element of type isotropic. Instead got {}".format(distribution))
```
To avoid unforseen/confusing errors (or worse, silently created erroneous objects), something like this can be added to all the `from_xml_element`methods in these two files. If you adopt this approach, then save some typing with a custom Exception class. Something like...

```
class UnexpectedElementError(Exception):
    """
    Raised when trying to create an object with an xml element type that doesn't match the type of the object.
    """
    def __init__(self, expected_type, unexpected_type):
        Exception.__init__(self, "Expected XML element of type {}. Instead got one of type {}".format(expected_type, unexpected_type))
```